### PR TITLE
Notification system for product evolution

### PR DESCRIPTION
**Pull request by bahati10**
### What does this PR do?
This PR implements a notification system to inform 
users about product lifecycle changes.

### Description of Task to be completed?

- Implemented a product notification system.
- The system sends email and in-app notifications to merchants upon product lifecycle changes (added, removed, expired, purchased, etc.).
- Utilizes Cron jobs for scheduling notification triggers.
- Leverages Node.js events to listen for different types of notification triggers based on product changes.
 ### How should this be manually tested?

- Simulate product lifecycle changes (adding, removing, expiring, purchasing products).
- Verify email notifications are sent to the merchant's email address associated with the product.
- Check if in-app notifications appear for the merchant upon product changes.

### What are the relevant pivotal tracker stories

#187570894

### CODE REVIEW

**Review by kirengamartial — APPROVED** [review]
the pull request code looks clean and ready for merge

**Review by Balinda21 — APPROVED** [review]
Well done everything is clear

File: src/handles/notifications.service.ts
Comment (Ishimwe7) [inline src/handles/notifications.service.ts:1]: everything is clear here.

**Review by Ishimwe7 — APPROVED** [review]
I think your codes are ready to be merged

**Review by derrick-nuby — CHANGES_REQUESTED** [review]
looks good and ready to merge

**Pull request merged**